src/topology/perfect.lean

```diff
@@ -0,0 +1,231 @@
+  exact ⟨x, ⟨xU, xC⟩⟩,
+end
+
+/-- Given a perfect nonempty set in a T2.5 space, we can find two disjoint perfect subsets
+This is the main inductive step in the proof of the Cantor-Bendixson Theorem-/
+lemma perf_nonempty.splitting [t2_5_space α] (hC : perfect C) (hnonempty : C.nonempty) :
+  ∃ C₀ C₁ : set α, (perfect C₀ ∧ C₀.nonempty ∧ C₀ ⊆ C) ∧
+  (perfect C₁ ∧ C₁.nonempty ∧ C₁ ⊆ C) ∧ disjoint C₀ C₁ :=
+begin
+  cases hnonempty with y yC,
+  obtain ⟨x, xC, hxy⟩ : ∃ x ∈ C, x ≠ y,
+  { have := hC.acc _ yC,
+    rw acc_pt_iff_nhds at this,
+    rcases this univ (univ_mem) with ⟨x,xC,hxy⟩,
+    exact ⟨x,xC.2,hxy⟩, },
+  obtain ⟨U, xU, Uop, V, yV, Vop, hUV⟩ := exists_open_nhds_disjoint_closure hxy,
+  use [closure (U ∩ C), closure (V ∩ C)],
+  split; rw ← and_assoc,
+  { refine ⟨hC.closure_nhd_inter x xC xU Uop, _⟩,
+    rw hC.closed.closure_subset_iff,
+    apply inter_subset_right, },
+  split,
+  { refine ⟨hC.closure_nhd_inter y yC yV Vop, _⟩,
+    rw hC.closed.closure_subset_iff,
+    apply inter_subset_right, },
+  apply disjoint.mono _ _ hUV; apply closure_mono; apply inter_subset_left,
+end
+
+section kernel
+
+/-- The Cantor-Bendixson Theorem: Any closed subset of a second countable space
+can be written as the union of a countable set and a perfect set.-/
+theorem countble_union_perfect_of_closed [second_countable_topology α]
+  (hclosed : is_closed C) :
+  ∃ V D : set α, (V.countable) ∧ (perfect D) ∧ (C = V ∪ D) :=
+begin
+  have := topological_space.exists_countable_basis α,
+  rcases this with ⟨b, bct, bnontrivial, bbasis⟩,
+  let v := {U ∈ b | (U ∩ C).countable},
+  let V := ⋃ U ∈ v, U,
+  let D := C \ V,
+  have Vct : (V ∩ C).countable,
+  { simp[V, Union_inter],
+    apply countable.bUnion,
+    { apply @countable.mono _ _ b,
+      { apply inter_subset_left, },
+      exact bct, },
+    apply inter_subset_right, },
+  refine ⟨V ∩ C, D, Vct, ⟨_, _⟩, _⟩,
+  { apply hclosed.sdiff,
+    apply is_open_bUnion,
+    rintros U ⟨Ub,-⟩,
+    exact is_topological_basis.is_open bbasis Ub, },
+  { rw preperfect_iff_nhds,
+    intros x xD E xE,
+    have : ¬ (E ∩ D).countable,
+    { intro h,
+      obtain ⟨U, hUb, xU, hU⟩ : ∃ U ∈ b, x ∈ U ∧ U ⊆ E,
+      { exact (is_topological_basis.mem_nhds_iff bbasis).mp xE, },
+      have : U ∈ v,
+      { use hUb,
+        apply @countable.mono _ _ ((E ∩ D) ∪ (V ∩ C)),
+        { rintros y ⟨yU, yC⟩,
+          by_cases y ∈ V,
+          { right,
+            exact ⟨h, yC⟩, },
+          left,
+          split,
+          { exact hU yU, },
+          exact ⟨yC, h⟩, },
+        apply countable.union h Vct, },
+      apply xD.2,
+      exact mem_bUnion this xU, },
+    by_contradiction,
+    push_neg at h,
+    apply this,
+    apply countable.mono h,
+    apply set.countable_singleton, },
+  rw[inter_comm, inter_union_diff],
+end
+
+/-- Any uncountable closed set in a second countable space contains a nonempty perfect subset.-/
+theorem perf_nonempty_of_closed_unctble [second_countable_topology α]
```

Comment:
name suggestion:
```suggestion
theorem exists_perfect_nonempty_of_is_closed_of_not_countable [second_countable_topology α]
```
We could also leave out `nonempty` from the name to make it shorter.
My reasons are: the result is an `exists`, `perf_nonempty` does not exist anymore and should be replaced, being closed is called `is_closed` and the hypothesis of not being countable should be denoted by `not_countable`.